cfme/rest/gen_data.py

```diff
@@ -487,6 +487,15 @@ def mark_vm_as_template(appliance, provider, vm_name):
     return appliance.rest_api.collections.templates.get(name=vm_name)
 
 
+def delete_templates_from_provider(provider, *template_names):
+    """Helper function to delete templates from provider with the help of wrapanapi.
+    """
+    for template in provider.mgmt.list_templates():
+        if template.name in template_names:
+            template.delete()
+    return True
```

Comment:
Why is `True` returned here?

Reply:
This is a costly way of doing this operation, since providers could have thousands of templates.  The `wrapanapi` `get_template` method should be used for this.

Second, I would argue against the inclusion of this function at all, since this is something that can be expressed concisely in-line in list comprehension.

Reply:
@mshriver I didn't realize we had a `get_template` method, guess I must've missed it somehow! And if this the case, we definitely do not need an external function for deleting templates.